src/index.js

```diff
@@ -2,12 +2,22 @@ import React from "react";
 import ReactDOM from "react-dom";
 import "./index.css";
 import App from "./App";
-import { ProductsProvider } from "./pages/products-context";
+import { ProductsProvider } from "./contexts/productsContext";
+import { BrowserRouter, Route, Routes } from "react-router-dom";
+import Product from "./pages/Product";
+import { CartProvider } from "./contexts/cartContext";
 
 ReactDOM.render(
     <React.StrictMode>
         <ProductsProvider>
-            <App />
+            <CartProvider>
+                <BrowserRouter>
+                    <Routes>
+                        <Route path="/" element={<App />} />
+                        <Route path="/product/:id" element={<Product />} />
```

Comment:
I gererally consider App has whole application and then declare routes inside it, not sure if this is good or bad practice.